src/blocks/video.js

```diff
@@ -9,11 +9,11 @@ module.exports = Block.extend({
   // more providers at https://gist.github.com/jeffling/a9629ae28e076785a14f
   providers: {
     vimeo: {
-      regex: /(?:http[s]?:\/\/)?(?:www.)?vimeo.com\/(.+)/,
+      regex: /(?:http[s]?:\/\/)?(?:www.)?vimeo\.co(?:.+(?:\/)([^\/].*)+$)/,
```

Comment:
Yeah, that'd do it. Interestingly, even though mines abit of a bodge it's got less steps in it (plus it'll handle https and http addresses without issue). 

I do see where you're going with it looking for numbers at the end of the string, should have twigged on that one earlier.